tests/utils_tests/test_http.py

```diff
@@ -61,14 +74,11 @@ def test_dict_with_sequence_of_bytes(self):
 
     def test_dict_with_bytearray(self):
         self.assertEqual(urlencode({'a': bytearray(range(2))}, doseq=True), 'a=0&a=1')
-        self.assertEqual(urlencode({'a': bytearray(range(2))}, doseq=False), 'a=%5B%270%27%2C+%271%27%5D')
+        self.assertEqual(urlencode({'a': bytearray(range(2))}, doseq=False), 'a=bytearray%28b%27%5Cx00%5Cx01%27%29')
```

Comment:
I know it's a bit weird but I think to keep it, since we want to ensure that the `doseq=False` behaviour is correct. (I guess that's covered elsewhere but having both means the test, and what hangs on the switch, is easy to read.)